### ISSUE TITLE
Menu without using anchors

### ISSUE THREAD

**Issue by richgcook** [alvarotrigo/fullPage.js#2755]
If you enable lockAnchors with `lockAnchors:true` although this removes the hash from the URL, which is the expected output, it also disables the ability to click the navigational dots. Shouldn't these still work? I basically just need to turn off the hash in the URL.

**Comment by richgcook** [comment]
I have my own menu setup `menu: 'section.home .navigation'` too by the way. Maybe I could hook into the moveTo function however I feel removing the hash only should be an option.

**Comment by alvarotrigo** [comment]
Working as expected: http://jsfiddle.net/97tbk/1924/

Closed until an isolated reproduction is provided as detailed in the [reporting issues guidelines](https://github.com/alvarotrigo/fullPage.js#reporting-issues) and the [issue skeleton](https://github.com/alvarotrigo/fullPage.js/blob/master/.github/ISSUE_TEMPLATE.md).

**Comment by richgcook** [comment]
Not sure what this jsFiddle is supposed to show but it doesn't have a custom menu set up?

**Comment by alvarotrigo** [comment]
I though you were talking about the horizontal navigation, the option `slidesNavigation:true`.
Menu won't work without anchors.

But you can always create your own menu and use the `$.fn.fullpage.moveTo` funcction by yourself.

**Comment by alvarotrigo** [comment]
The menu option allows you to link menu elements with any section or slide in the page.
If you do not want to use anchors, then you would have to use an `index` value to link to the menu. 

I guess I can add that feature. I'll mark as an enhancement.

**Comment by levelgain** [comment]
Hi can you expand on how and where this $.fn.fullpage.moveTo function would be included in my code?

I am trying to make my own navigation menu and I do not want to use anchors, which leave history in the user's browser. If they press 'back' I want it to just take the user away from the page, but I still want to have a navigation menu.

Thanks for any assistance.

**Comment by alvarotrigo** [comment]
> Hi can you expand on how and where this $.fn.fullpage.moveTo function would be included in my code?

Example [in the docs](https://github.com/alvarotrigo/fullPage.js#methods). Take a look at the demo.

Also take a look at my answer here:
https://stackoverflow.com/a/36742906/1081396

**Comment by levelgain** [comment]
Hi again. Many Thanks. I'm still finding that the items that have been clicked on from the menu are recorded in the browser history. The page doesn't scroll through them, but I have to click back several times to exit the page.

This is what I have here; any advice much appreciated.

<script type="text/javascript">
		$(document).ready(function() {
			$('#fullpage').fullpage({
                anchors: ['firstPage', 'secondPage', 'thirdPage', 'fourthPage', 'lastPage'],
                menu: '#myMenu',
                recordHistory: false,
				sectionsColor: ['#a9a9a9', '#4BBFC3', '#7BAABE' , '#1bbc9b', '#1bbc9b']
			});
		});
	</script>

<div id="myNav" class="overlay">
  <a href="javascript:void(0)" class="closebtn" onclick="closeNav()"></a>
  <div class="overlay-content"><ul id="myMenu">          
        <li data-menuanchor="firstPage" class="active"><a href="#firstPage" onclick="closeNav()">View My Work</a></li>
        <li data-menuanchor="secondPage"><a href="#secondPage" onclick="closeNav()">Extreme Brat Camp (Documentary)</a></li>
        <li data-menuanchor="thirdPage"><a href="#thirdPage" onclick="closeNav()">PhotoBox (Commercial)</a></li>
        <li data-menuanchor="fourthPage"><a href="#fourthPage" onclick="closeNav()">Two In A Million (Documentary)</a></li>
      <li data-menuanchor="fifthPage"><a href="#fifthPage" onclick="closeNav()">Emulsion (Feature Film)</a></li>
      <li data-menuanchor="sixthPage"><a href="#sixthPage" onclick="closeNav()">BBC 'This Is What We Do' (Commercial)</a></li>
      </ul>
  </div>
</div>
<div id="fullpage">
	<div class="section " id="section0"><div class="column1">
        <div class="headerbox"><div class="hb-left">
            <ul><li><h1 class="h-left"><bold>Extreme Brat Camp</bold><italic> (Feature Documentary)</italic></h1></li><li><h2 class="h-left">Ten Alps  / Brook Lapping.&emsp;&emsp;<italic>Dir:</italic> Andrew Parkin &emsp;&emsp;<italic>TX:</italic> Ch.4 (2014)</h2></li></ul></div></div>
        <div class="maincontentbox"><div class="embed-container"><iframe src="https://player.vimeo.com/video/108803751?title=0&byline=0&portrait=0" frameborder="0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe></div><div class="textcontentbox">This is some text about Extreme Brat Camp</div>
        </div>
        </div>
        <div class="column2"><div class="subcontentbox">This is some text about Extreme Brat Camp</div><div class="musiccontentbox"><div class="soundcloud-container"><iframe width="100%" height="100%" scrolling="no" frameborder="no" src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/playlists/131703705&amp;color=%23585c8c&amp;auto_play=false&amp;hide_related=false&amp;show_comments=true&amp;show_user=true&amp;show_reposts=false&amp;show_teaser=true"></iframe></div></div><div class="imagecontentbox"><div class="full-height-width"><img class="centerimage" src="/Media/16918025b.jpg"></div></div></div>
	</div>

**Comment by levelgain** [comment]
Hi again. Many Thanks. I'm still finding that the sections which have been clicked on from the menu are recorded in the browser history. The page doesn't scroll through them as I press back, but I have to click back several times to exit the page.

This is what I have; are there mistakes here?

`<script type="text/javascript" src="../jquery.fullPage.js"></script>
	<script type="text/javascript" src="examples.js"></script>
	<script type="text/javascript">
		$(document).ready(function() {
			$('#fullpage').fullpage({
                anchors: ['firstPage', 'secondPage', 'thirdPage', 'fourthPage', 'lastPage'],
                menu: '#myMenu',
                recordHistory: false,
				sectionsColor: ['#a9a9a9', '#4BBFC3', '#7BAABE' , '#1bbc9b', '#1bbc9b']
			});
		});
	</script>`

and 

`<div id="myNav" class="overlay">
  <a href="javascript:void(0)" class="closebtn" onclick="closeNav()"></a>
  <div class="overlay-content"><ul id="myMenu">          
        <li data-menuanchor="firstPage" class="active"><a href="#firstPage" onclick="closeNav()">View My Work</a></li>
        <li data-menuanchor="secondPage"><a href="#secondPage" onclick="closeNav()">Extreme Brat Camp (Documentary)</a></li>
        <li data-menuanchor="thirdPage"><a href="#thirdPage" onclick="closeNav()">PhotoBox (Commercial)</a></li>
        <li data-menuanchor="fourthPage"><a href="#fourthPage" onclick="closeNav()">Two In A Million (Documentary)</a></li>
      <li data-menuanchor="fifthPage"><a href="#fifthPage" onclick="closeNav()">Emulsion (Feature Film)</a></li>
      <li data-menuanchor="sixthPage"><a href="#sixthPage" onclick="closeNav()">BBC 'This Is What We Do' (Commercial)</a></li>
      </ul>
  </div>
</div>`

**Comment by levelgain** [comment]
Hi again. Many Thanks. I'm still finding that the sections which have been clicked on from the menu are recorded in the browser history. The page doesn't scroll back through them as I press back, but I have to click back several times to exit the page.

This is what I have; any advice much appreciated.

```<script type="text/javascript" src="../jquery.fullPage.js"></script>
	<script type="text/javascript" src="examples.js"></script>
	<script type="text/javascript">
		$(document).ready(function() {
			$('#fullpage').fullpage({
                sectionsColor: ['#a9a9a9', '#4BBFC3', '#7BAABE' , '#1bbc9b', '#1bbc9b'],
                anchors: ['firstPage', 'secondPage', 'thirdPage', 'fourthPage', 'lastPage'],
                menu: '#myMenu',
                recordHistory: false,
			});
		});
	</script>

```
```
<div id="myNav" class="overlay">
  <a href="javascript:void(0)" class="closebtn" onclick="closeNav()"></a>
  <div class="overlay-content"><ul id="myMenu">          
        <li data-menuanchor="firstPage" class="active"><a href="#firstPage" onclick="closeNav()">View My Work</a></li>
        <li data-menuanchor="secondPage"><a href="#secondPage" onclick="closeNav()">Extreme Brat Camp (Documentary)</a></li>
        <li data-menuanchor="thirdPage"><a href="#thirdPage" onclick="closeNav()">PhotoBox (Commercial)</a></li>
        <li data-menuanchor="fourthPage"><a href="#fourthPage" onclick="closeNav()">Two In A Million (Documentary)</a></li>
      <li data-menuanchor="fifthPage"><a href="#fifthPage" onclick="closeNav()">Emulsion (Feature Film)</a></li>
      <li data-menuanchor="sixthPage"><a href="#sixthPage" onclick="closeNav()">BBC 'This Is What We Do' (Commercial)</a></li>
      </ul>
  </div>
</div>
```

**Comment by alvarotrigo** [comment]
@levelgain  if you believe it is a bug please create a new issue and provide an isolated reproduction as detailed in the [reporting issues guidelines](https://github.com/alvarotrigo/fullPage.js#reporting-issues) and the [issue skeleton](https://github.com/alvarotrigo/fullPage.js/blob/master/.github/ISSUE_TEMPLATE.md).

Please use the minimum HTML code and no external JS and CSS. Get rid of everything else and ideally use empty sections.
This way I'll be sure the problem is not in your code and that it is indeed in fullpage.js.
It will also help me resolved your issue much faster.

**Comment by levelgain** [comment]
Ok sorry; I've tried to simplify my code. I'm sure there is no bug and that I have made an error; I just haven't worked it out. As I said, when I scroll there is no history recorded to the browser, but when I click on menu, sections are recorded to the browser history, which is what I'd like to avoid. Thanks again.

```
$(document).ready(function() {
			$('#fullpage').fullpage({
				sectionsColor: ['#1bbc9b', '#4BBFC3', '#7BAABE'],
                anchors: ['firstPage', 'secondPage', 'thirdPage'],
                menu: '#myMenu',
                recordHistory: false
```

```
<ul id="myMenu" style='position: fixed; z-index: 300;'>          
        <li data-menuanchor="firstPage" class="active"><a href="#firstPage">firstPage</a></li>
        <li data-menuanchor="secondPage"><a href="#secondPage">secondPage</a></li>
        <li data-menuanchor="thirdPage"><a href="#thirdPage">thirdPage</a></li>
      </ul>
```

**Comment by alvarotrigo** [comment]
I'll wait for the isolated reproduction.

**Comment by MadanBhandari** [comment]
As this is not yet supported, meanwhile you can get this by jQuery. 

When you have `lockAnchors: true,` , you can do this : 
 ```
       $('ul li a').click(function(){
          let newSlide = $(this).closest('li').data("menuanchor")
          $.fn.fullpage.moveTo(newSlide, 1);
        });
```

**Comment by alvarotrigo** [comment]
@MadanBhandari exactly :)

**Comment by alvarotrigo** [comment]
Fixed on the dev version 3.0.4.
It will get merged soon in the next release!

**Comment by alvarotrigo** [comment]
Now fixed in fullPage.js 3.0.4!
